SwrveSDK/src/main/java/com/swrve/sdk/SwrveBase.java

```diff
@@ -251,6 +239,43 @@ public void run() {
         return (T) this;
     }
 
+    private void setDefaultMessageListener() {
+        setMessageListener(new ISwrveMessageListener() {
+            public void onMessage(final SwrveMessage message) {
+                // Start a Conversation activity to display the campaign
+                if (SwrveBase.this.context != null) {
+                    final Context ctx = SwrveBase.this.context.get();
+                    if (ctx == null) {
+                        SwrveLogger.e(LOG_TAG, "Can't display a in-app message without a context");
+                        return;
+                    }
+
+                    Intent intent = new Intent(ctx, SwrveInAppMessageActivity.class);
+                    intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+                    intent.putExtra("inAppMessage", message);
```

Comment:
I really don't think we should be serializing a SwrveMessage to the activity. And I don't like that everything is serializable as a result. And also transient! It smell of poor design.

All those SwrveMessages are cached in SwrveImp someplace? Is there an id the activity can receive and look up directly in SwrveImp?